packages/auth/src/auth/auth.ts

```diff
@@ -96,24 +96,26 @@ export const init = async ({
   tidalAuthServiceBaseUri,
   tidalLoginServiceBaseUri,
 }: InitArgs) => {
-  const persistedUser = await loadCredentials(credentialsStorageKey);
+  const persistedCredentials = await loadCredentials(credentialsStorageKey);
 
   const credentials = {
-    ...persistedUser,
+    ...persistedCredentials,
     clientId,
     ...(clientSecret && {
       clientSecret,
     }),
     clientUniqueKey,
     credentialsStorageKey,
+    // we store the clientSecret separately to determine if a token needs to be upgraded
+    previousClientSecret: persistedCredentials?.clientSecret,
```

Comment:
Not an elegant solution, but the best I could do without changing a lot.

Reply:
Are we storing the secret in encrypted storage? Strictly speaking only confidential clients should use the secret, but if we store it in encrypted storage, we are doing the best we can to protect it in case someone would like to use it for non-confidential clients as well.

Reply:
We store everything in the encrypted storage, so should be fine :)